File: tests/fixtures/_highlight-corpus/sample.java
// Java sample
package com.example.demo;

import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

public final class Counter {
    private int value;

    public Counter() {
        this.value = 0;
    }

    public int increment(int n) {
        this.value += n;
        return this.value;
    }

    public int value() {
        return this.value;
    }

    public static void main(String[] args) throws Exception {
        List<String> lines = Files.readAllLines(Path.of("input.txt"));
        var counter = new Counter();
        for (String line : lines) {
            if (!line.isEmpty()) {
                counter.increment(line.length());
            }
        }
        System.out.printf("total bytes: %d%n", counter.value());
    }
}

File: tests/fixtures/_highlight-corpus/sample.kt
// Kotlin sample
package demo

import java.io.File

class Counter(private var value: Int = 0) {
    fun increment(n: Int = 1): Int {
        value += n
        return value
    }
    fun value(): Int = value
}

fun main() {
    val counter = Counter()
    File("input.txt").useLines { lines ->
        for (line in lines) {
            if (line.isNotEmpty()) {
                counter.increment(line.length)
            }
        }
    }
    println("total bytes: ${counter.value()}")
}
